### ISSUE TITLE
Not receiving :delete events when watching directories with watch/add

### ISSUE THREAD

**Issue by R1ck77** [zcaudate/hara#27]
*Forewarn*: I'm very new about hara: I have found it specifically, while I was looking for something that could help me in watching files without dealing with the complexity of `WatchService`, so I have no previous experience with it.
 
**Setup**

I'm using a leiningen `new app` template, with the following dependencies:

    :dependencies [
                 [org.clojure/clojure "1.8.0"]
                 [im.chit/hara.common.watch "2.4.4"]
                 [im.chit/hara.io.watch "2.4.4"]]

and this code:

    (ns com.example.haratest                                                                                                                           
      (:require [hara.io.watch]                                                                                                                           
                [hara.common.watch :as watch])                                                                                                            
      (:gen-class))
     
    (defn- watch-test [dir]
      (watch/add (clojure.java.io/file dir)
                 :my-watch
                 (fn [obj key prev next]
                   (println "Something happened! " obj key prev next))
                 {
                  :types #{:create :modify :delete}}))

    (defn -main
      [& args]
      (watch-test "/tmp"))

**Result**

Creating or modifying a file yields the expected results:

    $ touch /tmp/bar
    $ touch /tmp/bar

will produce:

    Something happened!  #object[java.io.File 0x14ef8c20 /tmp] :my-watch nil [:create #object[java.io.File 0x617c5b7d /tmp/bar]]
    Something happened!  #object[java.io.File 0x14ef8c20 /tmp] :my-watch nil [:modify #object[java.io.File 0x797ba4b8 /tmp/bar]]

but removing a file (whether already present before watching or not, it makes no difference) doesn't produce any effect:

    $ rm /tmp/bar

will yield nothing

**Expected result**

removing a file should generate an event, something like this:

    Something happened!  #object[java.io.File 0x14ef8c20 /tmp] :my-watch nil [:delete #object[java.io.File 0x797ba4b8 /tmp/bar]]

I'm running Ubuntu (tested with two different versions), but running the java example

[Oracle page on WatchService](https://docs.oracle.com/javase/tutorial/essential/io/notification.html)
 
seems to work so I'm sort of confident that the problem is not Os related.

Again, I apologize for any naivety on my part, and thank you for the awesome work on the library (I look forward to use more of it, despite this minor setback).

**Comment by zcaudate** [comment]
hey Riccardo,

Thanks for such a detailed writeup. I'm currently a little preoccupied but
will look at it asap. I'm actually a little surprised at this behavior
because I'm pretty sure I checked it. I hope you can wait for a couple more
days.

Chris.
​

**Comment by R1ck77** [comment]
Thank you for the prompt reply, Chris!

Take your time: I've used the library in such uneducated way, that I'm quite sure that the issue will be my fault, in the end... ;-)

Have a nice week end!

**Comment by zcaudate** [comment]
you are right. the delete events are not registering on my computer as well.

I have to go into this.

**Comment by R1ck77** [comment]
Thank you for the heads up (and glad that I did not waste your time with a setup on my side)!

**Comment by zcaudate** [comment]
okay, I have the reason.

this `if` block is making a check whether the file `.isFile` and so on delete events, it does not get processed. https://github.com/zcaudate/hara/blob/master/src/hara/io/watch.clj#L76

I'll make the release a little bit later but in the meantime. you can patch the code by cloning the repo, commenting  out the `if` clause:

```clojure
        ;;(if (.isFile file))
        (process-event watcher kind file)
```

change the project version to 2.4.5 and then running:

`lein repack install`

**Comment by R1ck77** [comment]
That was quick :)

I'll do just that as soon as I'll work on it again (2am here... :) ): thank you very much, and keep up the great work with your library!

**Comment by zcaudate** [comment]
=) no worries. ping if there are any issues

**Comment by zcaudate** [comment]
the fix is up in 2.4.6

**Comment by R1ck77** [comment]
Thank you again!
